Dockerfile.devel

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+ARG GOLANG_VERSION=x.x.x
+FROM golang:${GOLANG_VERSION}
```

Comment:
this is the same Dockerfile, so there's no benefit at all of basing of `k8s-test-infra` image.

Reply:
OK. I would expect maybe the following to be in the `k8s-infra` image since these really are `common`:
```
RUN go install github.com/golangci/golangci-lint/cmd/golangci-lint@${GOLANGCI_LINT_VERSION} \
    && go install github.com/matryer/moq@${MOQ_VERSION}
```
But if you want to have me include them here instead of putting them in that image, that is OK to I guess.

Reply:
that's a good point, yeah for this Repo, we only need to be specific with CLIENT_GEN_VERSION and CONTROLLER_GEN_VERSION, the others can be central from `k8s-test-infra`

Reply:
once https://github.com/NVIDIA/k8s-test-infra/pull/19 is in, I'll update here to move this PR forward